src/Core/include/OpenLoco/Core/Timer.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <chrono>
+#include <cstdint>
+
+namespace OpenLoco::Core
+{
+    class Timer
+    {
+        using ClockType = std::chrono::high_resolution_clock;
+        using TimePointType = std::chrono::time_point<ClockType>;
+
+        TimePointType _start = ClockType::now();
+
+    public:
+        // Resets the timer
+        void reset()
+        {
+            _start = ClockType::now();
+        }
+
+        // Returns elapsed time in milliseconds
+        float elapsed() const
+        {
+            const auto now = ClockType::now();
+            const auto elapsed = std::chrono::duration_cast<std::chrono::nanoseconds>(now - _start);
```

Comment:
Can't you just do `return std::chrono::duration<double, std::milli>(now - _start)`

Reply:
That returns duration

Reply:
yeah but then do .count on it. https://godbolt.org/z/5rW4zWvfe